### PR TITLE
[ci] migrate docs build to github actions

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1442 

Description: migrate docs build to github actions

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
There are some warnings in the logs. logo is missing

**Comment by ydcjeff** [comment]
@vfdev-5 does it look good to you?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          pip install torch torchvision -f https://download.pytorch.org/whl/cpu/torch_stable.html -U
+          pip install -r requirements-dev.txt
+          pip install -r docs/requirements.txt
+          pip install git+https://github.com/vfdev-5/sphinxcontrib-versioning.git
+
+      - name: build docs
+        shell: bash -l {0}
+        run: sphinx-versioning --use-master-conf --use-master-templates build --greatest-tag --whitelist-branches master docs/source docs/build/html
+
+      # - name: deploy docs
```
Comment (vfdev-5) [inline .github/workflows/docs.yml:33]: Deploy part should be uncommented and have if condition on pushed to master or on tag, right ?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: install docs reqs
+        shell: bash -l {0}
+        run: |
+          pip install -U pip wheel
```
Comment (vfdev-5) [inline .github/workflows/docs.yml:23]: I was also thinking to replace installation part with the one we do it with conda and cache... Your cache PR is still open, let me comment out there what have to be done...

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @ydcjeff

File: .github/workflows/docs.yml (hunk 1 above)
Comment (ydcjeff) [reply .github/workflows/docs.yml:33]: There is `on: push - branches: [master]` and `tag: ["v*"]`. Yes, it should be uncommented, but no need for if condition, I think. Can you verify `tag` will work as I haven't used `tag` before?

File: .github/workflows/docs.yml (hunk 2 above)
Comment (ydcjeff) [reply .github/workflows/docs.yml:23]: I would suggest using `actions/setup-python` as it's easier for caching. But it's up to you, conda or pip

Comment (vfdev-5) [reply .github/workflows/docs.yml:23]: I agree with you and OK let's keep it simple here with `actions/setup-python`.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (vfdev-5) [reply .github/workflows/docs.yml:33]: Yes, you are right about push and tag. Probably, should work like that (to verfiy in the playground). My point is about pull request. Actually, I forgot that we wanted to build docs with netlify, so we can remove `pull_request:
branches: [master]` from here, I think. Thus, we build and deploy on push-master and on tag. PR's preview is done on netlify and it was also a nice to have from https://github.com/pytorch/ignite/issues/1270#issuecomment-703180980 and the idea is to refactor deps installation and docs building into 2 scripts that could be used here and in netlify. Could you please create those 2 scripts inside `.github/workflows` and reuse them here ?

Comment (ydcjeff) [reply .github/workflows/docs.yml:33]: I see!
That line should be uncommented, just want to test via PR if docs can be built and yes `pull_request` needs to be removed.

Comment (ydcjeff) [reply .github/workflows/docs.yml]: @vfdev-5 when I tested on my fork, `tags` works if you do with `git push --tags` and make a tag that starts with `v` from github UI. `release` only works if you make a release/tag from github UI.

Comment (vfdev-5) [reply .github/workflows/docs.yml]: Both are OK, I think

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        shell: bash -l {0}
+        run: bash .github/workflows/install_docs_deps.sh
+
+      - name: Build docs
+        shell: bash -l {0}
+        run: bash .github/workflows/build_docs.sh
+
+      - name: deploy docs
+        uses: peaceiris/actions-gh-pages@v3
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (vfdev-5) [inline .github/workflows/docs.yml:28]: Indent is wrong here

**Review by vfdev-5 — APPROVED** [review]
@ydcjeff thanks for the PR !

**Pull request merged**